stock_indicators/indicators/common/candles.py

```diff
@@ -71,7 +67,12 @@ def is_bearish(self) -> bool:
         return self.Close < self.Open
 
 
-CandleProperties = generate_cs_inherited_class(_CandleProperties, CsCandleProperties, "CandleProperties")
+class CandleProperties(generate_cs_inherited_class(_CandleProperties, CsCandleProperties)):
```

Comment:
If your intention here is to eventually discontinue exposing `CsCandleProperties` as a public property, it might make sense to remove the `... as CsCandleProperties` from the base import in `src/cslib/lib/__init__.py` and elsewhere and then remake this as a [@deprecated("Rename to 'CandleProperties`")](https://peps.python.org/pep-0702/) with name `CsCandleProperties` for intermediate backwards compatibility ... and then remove it entirely as a breaking change in the next major version.

Reply:
Though, I could be missing an important convention with how you're consuming `_CandleProperties` and using inheritance.

Also, quick question; is there such a thing as _private_ and _public_ class decorators in Python?

Reply:
> Also, quick question; is there such a thing as private and public class decorators in Python?

No, there's only conventional way to mark it private or public.

Reply:
@DaveSkender 
> Generally, I think it makes sense to only prepend the Cs when it's a native .NET type, for clarity. And, when it's a custom type only using an alias when there's a conflicting native Python type name.

Until now, I prepend `Cs` prefix when it's a object managed by `pythonnet` which comes from .NET runtime at least in my thought. These objects have methods and fields that always need .NET runtime to use them. That's why I prepend 'Cs'. Even they are not shown by IDE linting. `Cs` is mark for it.


> If your intention here is to eventually discontinue exposing CsCandleProperties as a public property, it might make sense to remove the ... as CsCandleProperties from the base import in src/cslib/lib/__init__.py

And It's just for keeping inheritance relationship between Py`Quote` and Py`CandleProperties` while also keeping Py`CandleProperties` and Cs`CandleProperties`. Because .NET does not allow multiple inheritance, pythonnet also doesn't allow it. The workaround is `generate_cs_inherited_class ()`

I don't think it is critical change and it affects to user-side, because `CandleProperties` and `Quote` from `generate_cs_inherited_class ()` was class in the first place. The only difference is linting. Before I changed it, I was linted as `variable`, but not it is shown as `class`

 
<img width="467" alt="image" src="https://github.com/DaveSkender/Stock.Indicators.Python/assets/38756230/8ba4c592-9f6d-46e0-b3ac-f2c9169b5f97">

<img width="532" alt="image" src="https://github.com/DaveSkender/Stock.Indicators.Python/assets/38756230/fdc3fb7c-4b50-46c4-805d-d1d034b72931">

Reply:
Gotcha, no worries.  This is mostly educational for me.  I can see the `Cs` prefix being helpful (as related to Pythonnet) for an internal developer, but it's meaningless for an external user who doesn't need to care about Pythonet.  That's why I brought up _public_ vs _private_.  Anything related to the _public API_ should be meaningful to the _end user_.  It doesn't sound like Python is really setup like that.  Just a suggestion, no need to hold this up.

Reply:
Yeah, Python user can access anything in the library, but there's only coding convention to mark it. But we are providing detail docs and defined shortcuts only for "public" API in [`stock_indicators/__init__.py`](https://github.com/DaveSkender/Stock.Indicators.Python/blob/main/stock_indicators/__init__.py).

Always thanks for your review :)